app/scripts/views/ready.js

```diff
@@ -7,167 +7,166 @@
+    headerTitle: t('Account verified'),
+    readyToSyncText: t('You are now ready to use %(serviceName)s')
+  }
+};
 
   /*eslint-enable camelcase*/
 
-  const View = FormView.extend({
-    template: Template,
-    className: 'ready',
-
-    events: _.extend({}, FormView.prototype.events, {
-      'click .btn-continue': FormView.preventDefaultThen('continue'),
-      'click .btn-create-recovery-key': FormView.preventDefaultThen('createRecoveryKey'),
-      'click .btn-goto-account': FormView.preventDefaultThen('gotoSettings')
-    }),
-
-    initialize (options = {}) {
-      this._templateInfo = TEMPLATE_INFO[this.keyOfVerificationReason(options.type)];
-      this.type = options.type;
-    },
-
-    setInitialContext (context) {
-      context.set({
-        emailVerified: this.getSearchParam('secondary_email_verified') || this.getSearchParam('primary_email_verified'),
-        escapedEmailReadyText: this._getEscapedEmailReadyText(),
-        escapedHeaderTitle: this._getEscapedHeaderTitle(),
-        escapedReadyToSyncText: this._getEscapedReadyToSyncText(),
-        headerId: this._getHeaderId(),
-        isPasswordReset: this.isPasswordReset(),
-        isSync: this.relier.isSync(),
-        secondaryEmailVerified: this.getSearchParam('secondary_email_verified') || null,
-        showContinueButton: !! this.model.get('continueBrokerMethod'),
-      });
-    },
-
-    continue () {
+const View = FormView.extend({
+  template: Template,
+  className: 'ready',
+
+  events: _.extend({}, FormView.prototype.events, {
+    'click .btn-continue': FormView.preventDefaultThen('continue'),
+    'click .btn-create-recovery-key': FormView.preventDefaultThen('createRecoveryKey'),
+    'click .btn-goto-account': FormView.preventDefaultThen('gotoSettings')
+  }),
+
+  initialize (options = {}) {
+    this._templateInfo = TEMPLATE_INFO[this.keyOfVerificationReason(options.type)];
+    this.type = options.type;
+  },
+
+  setInitialContext (context) {
+    context.set({
+      emailVerified: this.getSearchParam('secondary_email_verified') || this.getSearchParam('primary_email_verified'),
+      escapedEmailReadyText: this._getEscapedEmailReadyText(),
+      escapedHeaderTitle: this._getEscapedHeaderTitle(),
+      escapedReadyToSyncText: this._getEscapedReadyToSyncText(),
+      headerId: this._getHeaderId(),
+      isPasswordReset: this.isPasswordReset(),
+      isSync: this.relier.isSync(),
+      secondaryEmailVerified: this.getSearchParam('secondary_email_verified') || null,
+      showContinueButton: !! this.model.get('continueBrokerMethod'),
+    });
+  },
+
+  continue () {
+    return Promise.resolve().then(() => {
       const { account, continueBrokerMethod } = this.model.toJSON();
       if (continueBrokerMethod && account) {
-        this.invokeBrokerMethod(continueBrokerMethod, account);
+        return this.invokeBrokerMethod(continueBrokerMethod, account)
+          .catch(err => this.displayError(err));
```

Comment:
Just checking, would there ever be any benefit to flattening this `.catch` to the top level rather than making it specific to `invokeBrokerMethod`? Say, if the destructuring assignment from `this.model.toJSON()` failed or something?

Reply:
Great point, I'll move it.